pinot-server/src/main/java/org/apache/pinot/server/api/resources/SegmentMetadataFetcher.java

```diff
@@ -128,8 +137,49 @@ public static String getSegmentMetadata(SegmentDataManager segmentDataManager, L
         indexStatus.put(RANGE_INDEX, INDEX_AVAILABLE);
       }
 
+      if (Objects.isNull(columnIndexContainer.getJsonIndex())){
+        indexStatus.put(JSON_INDEX, INDEX_NOT_AVAILABLE);
+      } else {
+        indexStatus.put(JSON_INDEX, INDEX_AVAILABLE);
+      }
+
       columnIndexMap.put(entry.getKey(), indexStatus);
     }
     return columnIndexMap;
   }
+
+  /**
+   * Get the JSON object containing star tree index details for a segment.
+   */
+  private static List<Map<String, Object>> getStartreeIndexForSegmentColumns(SegmentDataManager segmentDataManager) {
+    List<StarTreeV2> starTrees = segmentDataManager.getSegment().getStarTrees();
+    return starTrees != null ? getImmutableSegmentStartreeIndexes(starTrees) : null;
+  }
+
+  /**
+   * Helper to loop over star trees of a segment to create a map containing star tree details.
+   */
+  private static List<Map<String, Object>> getImmutableSegmentStartreeIndexes(List<StarTreeV2> starTrees){
```

Comment:
```suggestion
  private static List<Map<String, Object>> getStarTreeIndexes(List<StarTreeV2> starTrees){
```

Reply:
This is done

Reply:
This is resolved in a different way than suggested, please do take a look. @Jackie-Jiang